pkg/registry/core/rest/storage_core.go

```diff
@@ -99,7 +99,7 @@ type LegacyRESTStorage struct {
 	ServiceNodePortAllocator           rangeallocation.RangeRegistry
 }
 
-func (c LegacyRESTStorageProvider) NewLegacyRESTStorage(restOptionsGetter generic.RESTOptionsGetter) (LegacyRESTStorage, genericapiserver.APIGroupInfo, error) {
+func (c LegacyRESTStorageProvider) NewLegacyRESTStorage(restOptionsGetter generic.RESTOptionsGetter) (LegacyRESTStorage, genericapiserver.APIGroupInfo, func(), error) {
```

Comment:
I expected LegacyRESTStorageProvider to become a Destroyable that knows how to destroy all the sub-storages.

Reply:
`LegacyRESTStorageProvider` does not use pointer receiver so I can edit/extend a list of destroy functions from within `LegacyRESTStorageProvider.NewLegacyRESTStorage`. Would it be ok to have `LegacyRESTStorage` implement `Destroy` method instead?